### PR TITLE
Implementation of source terms for lines

### PR DESCRIPTION
**Pull request by TomFischer**
1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.2)
2. [x] Tests covering your feature were added?
3. [x] Any new feature or behavior change was documented?

### CODE REVIEW

File: ProcessLib/SourceTerms/LineSourceTermFEM.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+private:
+    ParameterLib::Parameter<double> const& _parameter;
+
+    IntegrationMethod const _integration_method;
+    std::vector<
+        IntegrationPointData<NodalRowVectorType>,
+        Eigen::aligned_allocator<IntegrationPointData<NodalRowVectorType>>>
+        _ip_data;
```
Comment (endJunction) [inline ProcessLib/SourceTerms/LineSourceTermFEM.h]: Changing type to
```suggestion
std::vector<NodalRowVectorType>> _ip_data;
```
would make the `IntegrationPointData` unnecessary...

**Comment by endJunction** [comment]
@HBShaoUFZ Could you have a look on the [docu](https://deploy-preview-2685--ogs.netlify.com/docs/benchmarks/heatconduction/heatconduction-line_source_term/), and possibly the benchmarks too, please?

File: ProcessLib/SourceTerms/LineSourceTermFEM.h (hunk 1 above)
Comment (TomFischer) [reply ProcessLib/SourceTerms/LineSourceTermFEM.h]: Indeed, good point. Thanks.

File: ProcessLib/SourceTerms/LineSourceTermFEM.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        ShapeMatricesType, ShapeFunction::NPOINTS, NUM_NODAL_DOF, GlobalDim>;
+
+    using NodalVectorType = typename LocalAssemblerTraits::LocalVector;
+    using NodalRowVectorType = typename ShapeMatricesType::NodalRowVectorType;
+
+public:
+    LineSourceTermLocalAssembler(
+        MeshLib::Element const& element,
+        std::size_t const local_matrix_size,
+        bool is_axially_symmetric,
```
[the remark this answers is not in the document]
Comment (TomFischer) [reply ProcessLib/SourceTerms/LineSourceTermFEM.h]: Fixed.

File: web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+Since the analytical solution has a singularity at $(x, y) = (0, 0)$ the
+analytical solution in paraview is generated as follow:
+```
+if (coordsX^2<0.0001 & coordsY^2<0.0001, temperature, -1/(4*asin(1))*ln(sqrt(coordsX^2+coordsY^2))
+```
+
+#### Results and evaluation
+
+- Comparison with analytical solution:
```
Comment (HBShaoUFZ) [inline web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc]: One recommendation: Here the temperature profile curve should be plotted at least once. Normal readers may not be familiar with the problem. They maybe confused as the error plot was directly shown.

File: web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+### Cylindrical domain
+
+The Poisson equation on cylindrical domain of heigth $1$ and radius
+$r=1$ is solved. In the following figure the geometry, partly semi-transparent,
+is sketched. Furthermore, the mesh resolution is shown in the cylindrical domain
+within the first quadrant of the coordinate system. In the second quadrant the
+simulated temperature distribution is depicted.
+
+{{< img src="../LineSourceTermFigures/temperature_distribution_line_source_term_in_cylinder.png" >}}
```
Comment (HBShaoUFZ) [inline web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc]: Here just add a line in the 3D figure, showing where the temperatures were compared. I guess it does not matter. Just to be clear.

File: web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc (hunk 3 above)
Comment (TomFischer) [reply web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc]: Thanks for the recommendation. I added the temperature profile.

File: web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc (hunk 4 above)
Comment (TomFischer) [reply web/content/docs/benchmarks/heatconduction/heatconduction-line_source_term.pandoc]: Added the line to the figure where the temperature profile is plotted.

**Pull request merged**